gbpservice/nfp/orchestrator/modules/device_orchestrator.py

```diff
@@ -708,9 +899,60 @@ def _prepare_device_data(self, device_info):
         device['network_function_id'] = network_function_id
 
         device['advance_sharing_interfaces'] = (
-                    self._get_advance_sharing_interfaces(device['id']))
+            self._get_advance_sharing_interfaces(device['id']))
         return device
 
+    def device_healthy(self, event):
+        nfp_context = event.data
+        device = nfp_context['network_function_device']
+        event_results = event.graph.get_leaf_node_results(event)
+        flag = True
+        for event_result in event_results:
+            if event_result.id == "DEVICE_UP":
+                if event_result.result == "SUCCESS":
+                    self._update_network_function_device_db(
+                        device, 'HEALTH_CHECK_COMPLETED')
+                else:
+                    self._create_event(event_id='DEVICE_NOT_UP',
+                               event_data=device,
+                               is_internal_event=True)
+                    self._update_network_function_device_db(device,
+                                                            'DEVICE_NOT_UP')
+                    self._controller.event_complete(event)
+                    return
+            elif event_result.id == "PLUG_INTERFACE" and event_result.result == "FAILED":
+                self._create_event(event_id='DEVICE_CONFIGURATION_FAILED',
+                                   event_data=device,
+                                   is_internal_event=True)
+                self._controller.event_complete(event)
+                return
+        nf_id = nfp_context['network_function']['id']
+        device_configured_event = self._controller.new_event(id='DEVICE_CONFIGURED',
```

Comment:
We are doing same for all the events, Can we move this logic to new function, which can get list of event_ids